src/main/java/mekanism/client/gui/button/GuiButtonImageMek.java

```diff
@@ -0,0 +1,58 @@
+package mekanism.client.gui.button;
+
+import javax.annotation.Nonnull;
+import mekanism.client.render.MekanismRenderer;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+/**
+ * From GuiButtonImage with a couple fixes and support for rendering differently when disabled
+ */
+@SideOnly(Side.CLIENT)
+public class GuiButtonImageMek extends GuiButton {
+
+    private final ResourceLocation resourceLocation;
+    private final int xTexStart;
+    private final int yTexStart;
+    private final int hoverOffset;
+    private final int disabledOffset;
+
+    public GuiButtonImageMek(int id, int x, int y, int width, int height, int offsetX, int offsetY, int hoverOffset, ResourceLocation resource) {
+        this(id, x, y, width, height, offsetX, offsetY, hoverOffset, 0, resource);
+    }
+
+    public GuiButtonImageMek(int id, int x, int y, int width, int height, int offsetX, int offsetY, int hoverOffset, int disabledOffset, ResourceLocation resource) {
+        super(id, x, y, width, height, "");
+        this.xTexStart = offsetX;
+        this.yTexStart = offsetY;
+        this.hoverOffset = hoverOffset;
+        this.disabledOffset = disabledOffset;
+        this.resourceLocation = resource;
+    }
+
+    @Override
+    public void drawButton(@Nonnull Minecraft mc, int mouseX, int mouseY, float partialTicks) {
+        if (this.visible) {
+            //Ensure the color gets reset. The default GuiButtonImage doesn't so other GuiButton's can have the color leak out of them
```

Comment:
`GuiButtonImage` doesn't even change the colour - though the base `GuiButton` does the colour reset before its render *shrug*

Reply:
The issue is *that* GuiButtonImage does not reset the color, as GuiButton does not reset the color after drawing the text. So if you hover over a GuiButton then all the GuiButtonImages will have their color change due to the text's color leaking. As we need our own class anyways for being able to disable the image it is not that big a deal.